### PR TITLE
Add sw and ci for gemmx on narrow

### PR DESCRIPTION
**Pull request by xiaoling-yi**
In this PR, we add the software test for the GemmX accelerator(hooked up on the narrow ports, with normal data layout NHWC).

### CODE REVIEW

File: target/snitch_cluster/sw/apps/snax-streamer-gemm-conv-simd/data/datagen.py [hunk 1]
```diff
@@ -0,0 +1,668 @@
...
+                im2col_matrix[b, oh * out_width + ow, :] = input_region.reshape(-1)
+
+    im2col_matrix = im2col_matrix.reshape(batch_size * out_height * out_width, -1)
+    im2col_kernel = kernel.reshape(out_channels, -1).T
+
+    return im2col_matrix, im2col_kernel
+
+
+# Golden model in python
+# Golden model in python
```
Comment (rgantonio) [inline target/snitch_cluster/sw/apps/snax-streamer-gemm-conv-simd/data/datagen.py:110]:
```suggestion

```

Double comment

File: target/snitch_cluster/sw/apps/snax-streamer-gemm-conv-simd/data/datagen.py [hunk 2]
```diff
@@ -0,0 +1,668 @@
...
+    tempLoop1,
+    spatial_len_0,
+    spatial_len_1,
+    tempStride0,
+    tempStride1,
+    spatialStride0,
+    spatialStride1,
+    data,
+    int32=False,
+):
```
Comment (rgantonio) [inline target/snitch_cluster/sw/apps/snax-streamer-gemm-conv-simd/data/datagen.py:148]: I think at this point, we need to make the data reshuffles or whatever useful functions in a single python script to make it easy for you 😄

File: target/snitch_cluster/sw/apps/snax-streamer-gemm-conv-simd/data/datagen.py [hunk 3]
```diff
@@ -0,0 +1,668 @@
...
+
+    # stride settings
+    strides = {
+        "M": tempStride1,
+        "K": tempStride0,
+        "m": spatialStride1,
+        "k": spatialStride0,
+    }
+
+    if int32:
```
Comment (rgantonio) [inline target/snitch_cluster/sw/apps/snax-streamer-gemm-conv-simd/data/datagen.py:167]: Why do you have a mode for int 32?

File: target/snitch_cluster/sw/apps/snax-streamer-gemm-conv-simd/data/datagen.py (hunk 2 above)
Comment (rgantonio) [reply target/snitch_cluster/sw/apps/snax-streamer-gemm-conv-simd/data/datagen.py:148]: I meant next PR you do that I guess? Or do you have more to fix so far?

**Review by rgantonio — APPROVED** [review]
Ok! 

Minor comment but maybe in the next PRs or so, but it looks like several are like "repeated" makefiles, codes and stuff, might be more elegant if we try to clean it soon. Before the number of items to test blows up!

Otherwise go!

How much more tests do you need to do?

**Comment by xiaoling-yi** [comment]
> Ok!
> 
> Minor comment but maybe in the next PRs or so, but it looks like several are like "repeated" makefiles, codes and stuff, might be more elegant if we try to clean it soon. Before the number of items to test blows up!
> 
> Otherwise go!
> 
> How much more tests do you need to do?

Yes! I will remove the repeated Python golden model code in the next PR!

I plan to have one more pooling layer and one precision-scalable test.

**Pull request merged**